ios/Fabric/RNCPagerViewComponentView.mm

```diff
@@ -0,0 +1,366 @@
+    switch (dismissKeyboard) {
+        case RNCViewPagerKeyboardDismissMode::None:
+            dismissKeyboardMode = UIScrollViewKeyboardDismissModeNone;
+            break;
+        case RNCViewPagerKeyboardDismissMode::OnDrag:
+            dismissKeyboardMode = UIScrollViewKeyboardDismissModeOnDrag;
+            break;
+    }
+    scrollView.keyboardDismissMode = dismissKeyboardMode;
+}
+
+
+- (void)updateProps:(const facebook::react::Props::Shared &)props oldProps:(const facebook::react::Props::Shared &)oldProps{
+    const auto &oldScreenProps = *std::static_pointer_cast<const RNCViewPagerProps>(_props);
+    const auto &newScreenProps = *std::static_pointer_cast<const RNCViewPagerProps>(props);
+    
+    // change index only once
+    if (_currentIndex == -1) {
+        _currentIndex = newScreenProps.initialPage;
+        [self shouldDismissKeyboard: newScreenProps.keyboardDismissMode];
+    }
+    
+    const auto newLayoutDirectionStr = RCTNSStringFromString(toString(newScreenProps.layoutDirection));
+    
+    
+    if (self.layoutDirection != newLayoutDirectionStr) {
+        self.layoutDirection = newLayoutDirectionStr;
+    }
+    
+    if (oldScreenProps.keyboardDismissMode != newScreenProps.keyboardDismissMode) {
+        [self shouldDismissKeyboard: newScreenProps.keyboardDismissMode];
+    }
+    
+    if (newScreenProps.scrollEnabled != scrollView.scrollEnabled) {
+        scrollView.scrollEnabled = newScreenProps.scrollEnabled;
+    }
+    
+    if (newScreenProps.overdrag != _overdrag) {
+        _overdrag = newScreenProps.overdrag;
+    }
+    
+    
+    
+    [super updateProps:props oldProps:oldProps];
+}
+
+- (void)handleCommand:(const NSString *)commandName args:(const NSArray *)args {
+    RCTRNCViewPagerHandleCommand(self, commandName, args);
+}
+
+#pragma mark - Internal methods
+
+- (void)setPage:(NSInteger)index {
+    [self goTo:index animated:YES];
+}
+
+- (void)setPageWithoutAnimation:(NSInteger)index {
+    [self goTo:index animated:NO];
+}
+
+- (void)goTo:(NSInteger)index animated:(BOOL)animated {
+    NSInteger numberOfPages = _nativeChildrenViewControllers.count;
+    
+    
+    if (numberOfPages == 0 || index < 0 || index > numberOfPages - 1) {
+        return;
+    }
+    
+    BOOL isForward = (index > self.currentIndex && [self isLtrLayout]) || (index < self.currentIndex && ![self isLtrLayout]);
+    UIPageViewControllerNavigationDirection direction = isForward ? UIPageViewControllerNavigationDirectionForward : UIPageViewControllerNavigationDirectionReverse;
+    
+    [self setPagerViewControllers:index
+                        direction:direction
+                         animated:animated
+         shouldCallOnPageSelected:YES];
+    
+}
+
+- (void)setPagerViewControllers:(NSInteger)index
+                      direction:(UIPageViewControllerNavigationDirection)direction
+                       animated:(BOOL)animated
+       shouldCallOnPageSelected:(BOOL)shouldCallOnPageSelected {
```

Comment:
I think, we can remove that parameter and document, that progress should be interpolated by reanimated. It would be nice to provide this behaviour in an example